extensions/blocks/opentable/opentable.php

```diff
@@ -0,0 +1,155 @@
+ *
+ * @return string
+ */
+function jetpack_opentable_block_load_assets( $attributes ) {
+	Jetpack_Gutenberg::load_assets_as_required( 'opentable' );
+
+	// The OpenTable script uses multiple `rid` paramters,
+	// so we can't use WordPress to output it, as WordPress attempts to validate it and removes them.
+	// phpcs:ignore WordPress.WP.EnqueuedResources.NonEnqueuedScript
+	return '<script type="text/javascript" src="' . esc_url( jetpack_opentable_build_embed_url( $attributes ) ) . '"></script>';
+}
+
+/**
+ * Get the a block attribute
+ *
+ * @param array $attributes Array of block attributes.
+ * @param array $attribute_name The attribute to get.
+ *
+ * @return string The filtered attribute
+ */
+function jetpack_opentable_block_get_attribute( $attributes, $attribute_name ) {
+	if ( isset( $attributes[ $attribute_name ] ) ) {
+		if ( in_array( $attribute_name, array( 'iframe', 'newtab' ), true ) ) {
+			return $attributes[ $attribute_name ] ? 'true' : 'false';
+		}
+		return wp_kses( $attributes[ $attribute_name ], array() );
+	}
+
+	$default_attributes = array(
+		'style'  => 'standard',
+		'iframe' => 'true',
+		'domain' => 'com',
+		'lang'   => 'en-US',
+		'newtab' => 'false',
+	);
+
+	return $default_attributes[ $attribute_name ];
+}
+
+/**
+ * Get the block type attribute
+ *
+ * @param array $attributes Array of block attributes.
+ *
+ * @return string The filtered attribute
+ */
+function jetpack_opentable_block_get_type_attribute( $attributes ) {
+	if ( count( $attributes['rid'] ) > 1 ) {
+		return 'multi';
+	}
+
+	if ( empty( $attributes['style'] ) || 'button' !== $attributes['style'] ) {
+		return 'standard';
+	}
+
+	return 'button';
+}
+
+/**
+ * Get the block theme attribute
+ *
+ * @param array $attributes Array of block attributes.
+ *
+ * @return string The filtered attribute
+ */
+function jetpack_opentable_block_get_theme_attribute( $attributes ) {
+	if ( empty( $attributes['style'] ) || 'button' === $attributes['style'] ) {
+		return 'standard';
+	}
+
+	return wp_kses( $attributes['style'], array() );
+}
+
+/**
+ * Build an embed URL from an array of block attributes.
+ *
+ * @param array $attributes Array of block attributess.
+ *
+ * @return string Embed URL
+ */
+function jetpack_opentable_build_embed_url( $attributes ) {
+	$url = add_query_arg(
+		array(
+			'type'   => jetpack_opentable_block_get_type_attribute( $attributes ),
+			'theme'  => jetpack_opentable_block_get_theme_attribute( $attributes ),
+			'iframe' => jetpack_opentable_block_get_attribute( $attributes, 'iframe' ),
+			'domain' => jetpack_opentable_block_get_attribute( $attributes, 'domain' ),
+			'lang'   => jetpack_opentable_block_get_attribute( $attributes, 'lang' ),
+			'newtab' => jetpack_opentable_block_get_attribute( $attributes, 'newtab' ),
+		),
+		'//www.opentable.com/widget/reservation/loader'
+	);
+
+	foreach ( $attributes['rid'] as $rid ) {
+		$url .= '&rid=' . wp_kses( $rid, array() );
+	}
+
+	return $url;
```

Comment:
What do you think about adding a filter here, allowing one to change the final URL and add extra parameters such as the marketing tracking options offered in the widget builder?

Reply:
✅